File: 프로그래머스/Lv.2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    var answer = 0;
    
    let a = arr[0]

    for (let i = 1; i < arr.length; i++) {
        b = LCM(a, arr[i])
        a = b
    }
    
    return a;
}

function GCD(x, y){
    while (y) {
        let z = x
        x = y
        y = z % y
    }
    return x
}

function LCM(x, y){
    return x * y / GCD(x, y)
}

File: 프로그래머스/Lv.2/12953. N개의 최소공배수/N개의 최소공배수.py
def solution(arr):
    arr.sort()
    answer = 0
    i = 1
    y = arr[i]
    
    while i <= len(arr):
        x = LCM(y, arr[i-1])
        y = x
        i += 1
        
    return y

def GCD (a, b):
    while b:
        a, b = b, a % b
    return a

def LCM (a, b):
    return a * b / GCD(a, b)
    